comid/tdx-profile/teeisvproid.go

```diff
@@ -0,0 +1,189 @@
+// nolint:dupl
+package tdx
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/veraison/corim/encoding"
+)
+
+// TeeISVProdID stores an ISV Product Identifier. The supported formats are uint and variable-length bytes.
+type TeeISVProdID struct {
+	val interface{}
+}
+
+// NewTeeISVProdID creates a new TeeISVProdID from the
+// supplied interface and return a pointer to TeeISVProdID
+// Supported values are positive integers and byte array
+func NewTeeISVProdID(val interface{}) *TeeISVProdID {
+	switch t := val.(type) {
+	case uint, uint64:
+		return &TeeISVProdID{val: t}
+	case []byte:
+		return &TeeISVProdID{val: t}
+	case int:
+		if t < 0 {
+			return nil
```

Comment:
nil as error value is an anti-pattern. Can you not return (*TeeISVProdID, error)?

Reply:
Now corrected, Thanks for your review!